ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -47,50 +61,220 @@ public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfi
             throw new IllegalArgumentException("model id is null");
         }
 
+        // Initialize model type from config if not set
+        if (modelConfig != null) {
+            this.modelConfig = modelConfig;
+        }
+
         // Create input for the predictor
         Input input = new Input();
         input.add(DEFAULT_WARMUP_QUESTION);
         input.add(DEFAULT_WARMUP_CONTEXT);
+        input.add("0");
 
         // Run prediction to warm up the model
         predictor.predict(input);
     }
 
-    /**
-     * Checks if the model is configured for sentence highlighting.
-     *
-     * @param modelConfig The model configuration
-     * @return true if the model is configured for sentence highlighting, false otherwise
-     */
-    private boolean isSentenceHighlightingType(MLModelConfig modelConfig) {
-        if (modelConfig != null) {
-            return SENTENCE_HIGHLIGHTING_TYPE.equalsIgnoreCase(modelConfig.getModelType());
-        }
-        return false;
-    }
-
     @Override
     public ModelTensorOutput predict(String modelId, MLInput mlInput) throws TranslateException {
         MLInputDataset inputDataSet = mlInput.getInputDataset();
-        List<ModelTensors> tensorOutputs = new ArrayList<>();
-        Output output;
         QuestionAnsweringInputDataSet qaInputDataSet = (QuestionAnsweringInputDataSet) inputDataSet;
         String question = qaInputDataSet.getQuestion();
         String context = qaInputDataSet.getContext();
+
+        if (isStandardQAModel()) {
+            return predictStandardQA(question, context);
+        }
+
+        return predictSentenceHighlightingQA(question, context);
+    }
+
+    private boolean isStandardQAModel() {
```

Comment:
naming isn't doesn't seem right. How about `isNotSentenceHighlightingModel`?

Reply:
Updated to use `isSentenceHighlightingModel`